src/features/evaluations/[id]/EvaluationsCriteria.tsx

```diff
@@ -0,0 +1,280 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+import { Icon } from "../../../components/icon/Icon"
+import { Button } from "../../../components/button/Button"
+import { getEvaluationTemplateContents } from "../../../redux/slices/evaluationTemplateContentsSlice"
+import { useAppDispatch } from "../../../hooks/useAppDispatch"
+import { useAppSelector } from "../../../hooks/useAppSelector"
+import { AnswerOptions } from "../../../types/answerOptionType"
+import {
+  submitAnswer,
+  submitComment,
+  submitEvaluation,
+  getUserEvaluations,
+} from "../../../redux/slices/userSlice"
+import { Loading } from "../../../types/loadingType"
+import { setAlert } from "../../../redux/slices/appSlice"
+import {
+  EvaluationStatus,
+  type Evaluation,
+} from "../../../types/evaluationType"
+import { formatDate } from "../../../utils/formatDate"
+import { TextArea } from "../../../components/textarea/TextArea"
+import { getAnswerOptionVariant } from "../../../utils/variant"
+
+export const EvaluationsCriteria = () => {
+  const { id, evaluation_id } = useParams()
+  const appDispatch = useAppDispatch()
+  const { evaluation_template_contents } = useAppSelector(
+    (state) => state.evaluationTemplateContents
+  )
+  const { loading, loading_answer, loading_comment, user_evaluations } =
+    useAppSelector((state) => state.user)
+  const [evaluation, setEvaluation] = useState<Evaluation>()
+  const [comments, setComments] = useState<string>("")
+
+  useEffect(() => {
+    if (evaluation_id !== "all") {
+      void appDispatch(
+        getEvaluationTemplateContents({
+          evaluation_id,
+        })
+      )
+      setEvaluation(
+        user_evaluations?.find(
+          (evaluation) => evaluation.id === parseInt(evaluation_id as string)
+        )
+      )
+    }
+  }, [evaluation_id, user_evaluations])
+
+  useEffect(() => {
+    if (evaluation?.comments !== undefined && evaluation?.comments !== null) {
+      setComments(evaluation.comments)
+    } else {
+      setComments("")
+    }
+  }, [evaluation])
+
+  const handleOnClickStar = async (
+    answerOptionId: number,
+    evaluationRatingId: number
+  ) => {
+    if (evaluation_id !== undefined) {
+      try {
+        const result = await appDispatch(
+          submitAnswer({
+            evaluation_id: parseInt(evaluation_id),
+            evaluation_rating_id: evaluationRatingId,
+            answer_option_id: answerOptionId,
+          })
+        )
+        if (result.payload.id !== undefined && id !== undefined) {
+          void appDispatch(
+            getEvaluationTemplateContents({
+              evaluation_id,
+            })
+          )
+          void appDispatch(
```

Comment:
same

Reply:
For here as well, so that when user submits, the status badge updates to submitted as well

Reply:
just update the evaluation status directly 🤔